ports/nrf/common-hal/_bleio/Adapter.c

```diff
@@ -594,7 +595,26 @@ STATIC void check_data_fit(size_t data_len, bool connectable) {
     }
 }
 
-uint32_t _common_hal_bleio_adapter_start_advertising(bleio_adapter_obj_t *self, bool connectable, float interval, uint8_t *advertising_data, uint16_t advertising_data_len, uint8_t *scan_response_data, uint16_t scan_response_data_len) {
+STATIC bool advertising_on_ble_evt(ble_evt_t *ble_evt, void *self_in) {
+    bleio_adapter_obj_t *self = (bleio_adapter_obj_t*)self_in;
+
+    switch (ble_evt->header.evt_id) {
+        case BLE_GAP_EVT_ADV_SET_TERMINATED:
+            mp_printf(&mp_plat_print, "Advertising set terminated - %d advertising events were completed - reason: %d\n", ble_evt->evt.gap_evt.params.adv_set_terminated.num_completed_adv_events, ble_evt->evt.gap_evt.params.adv_set_terminated.reason);
+            common_hal_bleio_adapter_stop_advertising(self);
+            ble_drv_remove_event_handler(advertising_on_ble_evt, self_in);
+            break;
+
+        default:
+            // For debugging.
+            mp_printf(&mp_plat_print, "Unhandled Advertising etvent: 0x%04x\n", ble_evt->header.evt_id);
```

Comment:
Also debugging (and it says "etvent" instead of "event")

Reply:
I corrected the spelling on this one and commented it out, like the others.